lib/oli/resources/alternatives/user_section_preference_strategy.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Oli.Resources.Alternatives.UserSectionPreferenceStrategy do
+  alias Oli.Resources.Alternatives.AlternativesStrategyContext
+  alias Oli.Delivery.ExtrinsicState
+  alias Oli.Resources.Alternatives.Selection
+
+  require Logger
+
+  @behaviour Oli.Resources.Alternatives.AlternativesStrategy
+
+  @doc """
+  Selects the first alternative that matches a given user preference.
+
+  If a preference is not set for a user, then the first alternative is selected.
+
+  In the rare case when there are no alternatives set, this will return an empty list `[]]`.
+  """
+  def select(
+        %AlternativesStrategyContext{user: user, section_slug: section_slug, mode: :delivery},
+        %{
+          "children" => children,
+          "alternatives_id" => alternatives_id
+        }
+      ) do
+    pref_key = ExtrinsicState.Key.alternatives_preference(alternatives_id)
+
+    case ExtrinsicState.read_section(
+           user.id,
+           section_slug,
+           MapSet.new([pref_key])
+         ) do
+      {:ok, %{^pref_key => pref}} ->
+        # return all children with display: :none except for the alternative that matches the selected preference
+        Enum.map(children, fn alt ->
+          if alt["value"] == pref do
+            %Selection{alternative: alt}
+          else
+            %Selection{alternative: alt, hidden: true}
```

Comment:
User preference strategy actually returns all alternatives wrapped in a `%Selection{}` struct. This struct has a `hidden` field which tells the renderer which items to render with a `hidden` class so that only the user preferred (or default/first) alternative is initially shown. The preference selector adds/removes the `hidden` class as necessary when it is changed